spec/controllers/users/magic_links_controller_spec.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+describe Users::MagicLinksController, type: :controller do
```

Comment:
nice specs 👍 

however I am personnally not a big fan of controller specs. 

My rationale for specs is : 
- only test at the highest possible level == E2E , but only test happy paths for important features, or special cases that we want to be sure are covered
- all other spec levels are optionnal and should only be written when writing them eases the actual development or maintenance. This implies mostly unit tests I’d write for services with a somewhat complicated logic, or a lot of branches. 

the reasoning is that writing many tests at different levels often slows down further evolutions because it requires refactoring the specs. For example I’m very unhappy when I have to refactor a mock in a unit test because I changed the inner implementation of the class being tested. **I also don’t feel that TDD is helping me go faster and safer in most cases**.

This is extremeley subjective and absolutely not a constraint to follow in this project or any project. But for the time being, I’d rather we stick to this no-controller-specs rule and I’ll let the new dev decide how s·he wants to handle it in the future!

I have not yet taken the time to write it but I want to write a doc explaining the coding guidelines in this codebase, the caveats, the smelly bits etc

Reply:
in that particular case I feel that the logic to test is contained in the service and is already tested - which is why I extracted it in the first place I guess. 

I’m at ease with this controller code not being tested because it is quite boring code. We could however write a E2E test for the happy path of this specific feature - login with a magic link - because it is quite a core feature. This E2E spec already exists in `features/communes/sign_in_with_token_spec.rb`